web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,76 @@
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-00.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshot modal - default](img/1-01.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+![Offline Snapshot modal - selecting disks](img/1-02.jpg)
+In the disks field, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+![Offline Snapshot modal - disks customization](img/1-03.jpg)
+![Offline Snapshot modal - disks customized](img/1-04.jpg)
+
+![Offline Snapshotting status indication](img/1-05.jpg)
+![Offline Snapshotting status indication on click and kebab mode](img/1-06.jpg)
+Any virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “Stop snapshot” would be available.
+
+![Offline Restoring from snapshot status](img/1-07.jpg)
+Any virtual machine that is undergoing a restoration process will show a "Restoring" indication on the status, with more details and a "Stop restoring" action on a popover. 
+
+
+## VM page - snapshots tab
+
+![Offline Shanpshot tab](img/2-00.jpg)
+The snapshot list is located under the VM. A tab item in the horizontal navigation will be dedicated for snapshots. The snapshot list is ordered by  date as a default, making the "Scheduled" snapshots appear firts.
+The user can sort the list by name or date, clicking on the title will allow the user to switch sorting and decide if it's ascending or descending.  
+![Offline Snapshot - same modal as in VM list](img/2-01.jpg)
+Same modal as in VM list - Snapshot action
+![Offline Snapshot tab -  snapshot item details](img/2-02a.jpg)
+The user clicks the snapshot's name to view more information. 
+![Offline Snapshot tab - snapshot item details](img/2-02.jpg)
+Details view.
+
+![Offline Snapshot tab - kebab actions](img/2-03.jpg)
+Kebab actions.
+Clicking on the kebab menu on each row will open an action menu with several options:
+- Create VM Template
+- Create VM
+- Restore
+- Delete
+![Offline Snapshot tab - restoring action](img/2-04.jpg)
+Restoring VM from snapshot modal. When clicking the restore option in the action menu it opens the restore popup. The restore popup aligned to the [cloning](http://openshift.github.io/openshift-origin-design/web-console/knikubevirt/clone-vm/clone-vm) and [template](http://openshift.github.io/openshift-origin-design/web-console/knikubevirt/vm-templates/vm-templates) popups.
+
+
+## Scheduled snapshots
```

Comment:
@cshinn is using a date picker in some of his alerting stuff. Does it make sense to try to align here, or are the use cases vastly different?

Reply:
Unfortunately, the datepicker doesn't exist yet, but I agree that we should try to make date/time inputs consistent throughout the console. Esp. with respect to timezone support

Reply:
I think this is something we should track as a future enhancement but should stop this PR from being merged. I don't want to this to disappear though, what's the best way to track these types of updates? @cshinn @ncameronbritt any ideas?

Reply:
@Ranelim I think this is the last item to resolve before merging. I'd go ahead and create a follow up issue. Going to approve from my side in the meantime :)

Reply:
@Ranelim I'd agree, we'll just need to ensure that when the datepicker exists we see consistency with it. I'm not sure how important it is with our designs but definitely in what's been implemented. So the issue could cover following up with Dev to see what they have planned to update any instance that isn't consistent. We can probably assist in pointing out where it needs to be updated.

Reply:
Adding the [date/time picker issue here](https://github.com/openshift/openshift-origin-design/issues/228)